### PR TITLE
Stress test with `send_columns` for Rust & C++, improve C++ `TimeColumn` api

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Add `send_column` variant to plot stress tests for C++ and Rust.
* Small addition to C++ `TimeColumn`: `from_seconds`
* `from_times_nanoseconds` -> `from_nanoseconds`,

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7148?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/7148?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] If have noted any breaking changes to the log API in `CHANGELOG.md` and the migration guide

- [PR Build Summary](https://build.rerun.io/pr/7148)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: rerun_cpp/src/rerun/time_column.hpp [hunk 1]
```diff
@@ -72,23 +72,29 @@ namespace rerun {
             );
         }
 
-        /// Creates a sequence time column from an array of sequence points.
+        /// Creates a sequence time column from an array of nanoseconds.
```
Comment (teh-cmc) [inline rerun_cpp/src/rerun/time_column.hpp:75]:
```suggestion
        /// Creates a time column from an array of nanoseconds.

File: rerun_cpp/src/rerun/time_column.hpp [hunk 2]
```diff
@@ -72,23 +72,29 @@ namespace rerun {
...
-        ) {
-            return TimeColumn(
-                Timeline(std::move(timeline_name), TimeType::Time),
-                std::move(times_in_nanoseconds),
-                sorting_status
-            );
-        }
+        );
+
+        /// Creates a sequence time column from an array of seconds.
```
Comment (teh-cmc) [inline rerun_cpp/src/rerun/time_column.hpp:87]:
```suggestion
        /// Creates a time column from an array of seconds.

**Review by teh-cmc — CHANGES_REQUESTED** [review]
I haven't looked at the code for the batched stress-tests yet, but the results are problematic (might just be a display issue, I havent digged).

`main`:
```
$ pixi run rs-plot-dashboard --num-plots 10 --num-series-per-plot 5 --num-points-per-series 5000 --freq 1000
logged 50050 scalars over 1.000245178s (freq=50037.732Hz, expected=50000.000Hz, load=46.504%)
logged 50000 scalars over 1.000337259s (freq=49983.143Hz, expected=50000.000Hz, load=51.753%)
logged 50000 scalars over 1.000439929s (freq=49978.013Hz, expected=50000.000Hz, load=51.538%)
logged 50000 scalars over 1.000516588s (freq=49974.184Hz, expected=50000.000Hz, load=43.453%)
logged 49950 scalars over 1.000539157s (freq=49923.084Hz, expected=50000.000Hz, load=44.543%)
```

after:
```
$ pixi run rs-plot-dashboard --num-plots 10 --num-series-per-plot 5 --num-points-per-series 5000 --freq 1000
logged 250250 scalars over 1.000527382s (freq=250118.092Hz, expected=50000.000Hz, load=130.322%)
logged 250000 scalars over 1.000632318s (freq=249842.020Hz, expected=50000.000Hz, load=39.225%)
logged 250000 scalars over 1.000734313s (freq=249816.556Hz, expected=50000.000Hz, load=42.736%)
logged 250000 scalars over 1.000833488s (freq=249791.802Hz, expected=50000.000Hz, load=33.822%)
logged 249750 scalars over 1.000836593s (freq=249541.236Hz, expected=50000.000Hz, load=26.430%)
```

**Comment by Wumpf** [comment]
@teh-cmc good catch! Looking at it again I believe both what I had originally as well as `main` are wrong. And the same applies to Python.

If you check on the diff you'll find that they both counted logged scalars as:
```rust
total_num_scalars += num_series;
```
Whereas what I did was:
```rust
let num_series = args.num_plots * args.num_series_per_plot;
let mut scalars_per_tick = num_series * args.num_series_per_plot;
// ...
total_num_scalars += scalars_per_tick;
```
which in your example is factor 5 * 5 = 25 more which matches the numbers
So but the number of scalars per tick isn't exactly the number of series times the number of series per plots, it is just the number of series! Each series gets a scalar every tick (unless we do columns which we we do colum-batch-size times many)

Where did I have this code? Ofc from python.

**Comment by teh-cmc** [comment]
??? wth just happened

**Comment by Wumpf** [comment]
Oo

**Comment by Wumpf** [comment]
where did stuff go. It first showed merged to me,then just closed now merged again

**Comment by teh-cmc** [comment]
dunno i rebase on main and stuff just disappeared -- ive repushed the previous state

**Pull request closed without merging**